tests/agent_features/test_attributes_in_action.py

```diff
@@ -938,10 +939,27 @@ def test_none_type_routing_key_agent_attribute():
 @background_task()
 def test_enduser_id_attribute_api():
     """Validate enduser.id attribute set by API ends up on transaction events, error events, and traced errors."""
-    txn = current_transaction()
-    txn._add_agent_attribute("enduser.id", "1234")  # TODO Replace this with an API call
+    set_user_id("1234")
 
     try:
         raise ValueError()
     except Exception:
         notice_error()
+
+
+@reset_core_stats_engine()
+@validate_error_trace_attributes_outside_transaction(
+    callable_name(ValueError), exact_attrs={"user": {}, "intrinsic": {}, "agent": {}}
+)
+@validate_error_event_attributes_outside_transaction(exact_attrs={"user": {}, "intrinsic": {}, "agent": {}})
+@validate_attributes("agent", [], _forgone_agent_attributes)
+def test_enduser_id_attribute_api_outside_txn():
+    """Validate enduser.id attribute set by API ends up on transaction events, error events, and traced errors."""
+    set_user_id("1234")
+
+    try:
+        raise ValueError()
+    except Exception:
+        application_instance = application()
+        application_instance.notice_error()
```

Comment:
Is there a point to running this test outside a transaction with these validators? If we don't expect any data, should this just be a no harm test that only looks for a crash?